step/serialization.md

```diff
@@ -260,31 +270,31 @@ public class ClassSerializationClient {
 </copy>
 ```
 
-Now run `ClassSerializationClient`:
+Now run `SerializationClient`:
 
 ```
 <copy>
-java ClassSerializationClient.java
+java SerializationClient.java
 </copy>
 ```
 
-Which will throw a NullPointerException, when in `ClassSerializationServer` `toUpperCase()` is called: 
+This will cause `SerializationServer` to throw a `NullPointerException` and a very confused developer as it should be possible for the field `message` to be null! 
 
 ```no-highlight
 Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.toUpperCase()" because the return value of "Object.toString()" is null
-	at ClassSerializationServer.main(ClassSerializationServer.java:32)
+	at SerializationServer.main(SerializationServer.java:32)
 ``` 
 
-This happens because in Java when a stream is being deserialized, the constructor for the class is not called and the fields are set reflectively. So even though in the version of `MessageClass` being used by `ClassSerializationServer` it should programmatically be impossible for `message` to be null. Serialization is able to break encapsulation and provide a null value, creating a difficult to track down bug. 
+This happens because in Java when a stream is being deserialized, the constructor for the class is not called and the fields are set reflectively. So even though in the version of `Message` being used by `SerializationServer` it should programmatically be impossible for `message` to be null. Serialization is able to break encapsulation and set the `message` field to null, creating a difficult to track down bug. 
 
 ### Injecting Code with Serialization
-The encapsulation breaking behavior of serialization can be problemmatic, but the real security risk can come from the ability to inject code via serialization. Let's take a look at an example of a [gadget chain](https://blog.redteam-pentesting.de/2021/deserialization-gadget-chain/) attack.
+The encapsulation breaking behavior of serialization is problemmatic, and the above example is but one simple example of how to break it. However the real security risk primarily comes from the ability to inject code via serialization. Let's take a look at an example of a [gadget chain](https://blog.redteam-pentesting.de/2021/deserialization-gadget-chain/) attack.
```

Comment:
However the real security risk primarily comes -> breaking encapsulation is also a security risk